api/client/service/opts.go

```diff
@@ -176,61 +263,28 @@ func (m *MountOpt) Set(value string) error {
 		}
 	}
 
-	for _, field := range fields {
-		parts := strings.SplitN(field, "=", 2)
-		if len(parts) == 1 && strings.ToLower(parts[0]) == "writable" {
-			mount.Writable = true
-			continue
+	var err error
+	switch strings.ToLower(key) {
+	case "name":
+		mount.Source = value
+	case "populate":
+		volumeOptions().Populate, err = strconv.ParseBool(value)
+		if err != nil {
+			return false, fmt.Errorf("invalid value for populate: %s", value)
 		}
-
-		if len(parts) != 2 {
-			return fmt.Errorf("invalid field '%s' must be a key=value pair", field)
+	case "label":
+		setValueOnMap(volumeOptions().Labels, value)
+	case "driver":
+		volumeOptions().DriverConfig.Name = value
+	case "driver-opt":
```

Comment:
Because there are options for volumes (write mask, propagation, etc), and then there are options to pass to a volume driver.